java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -229,6 +234,19 @@ public int hashCode() {
         .map(f -> new ExecutionPath(edge, visited.add(edge), newTrackSymbols, f, endOfPath));
     }
 
+    private Set<LearnedConstraint> filterRedundantObjectDomain(Set<LearnedConstraint> learnedConstraints) {
+      Map<SymbolicValue, Long> constraintsBySV = learnedConstraints.stream()
+        .filter(lc -> lc.constraint() != null)
+        .collect(Collectors.groupingBy(LearnedConstraint::symbolicValue, Collectors.counting()));
+      return learnedConstraints.stream()
+        .flatMap(lc -> isConstraintFromObjectDomain(lc.constraint()) && constraintsBySV.get(lc.symbolicValue()) > 1 ? Stream.empty() : Stream.of(lc))
+        .collect(Collectors.toSet());
+    }
+
+    private boolean isConstraintFromObjectDomain(@Nullable Constraint constraint) {
+      return constraint != null && constraint.getClass().isAssignableFrom(ObjectConstraint.class);
```

Comment:
why not simply use `constraint instanceof ObjectConstraint` ? if null, it will return `false`.